src/python_testing/matter_testing_infrastructure/chip/testing/matter_testing.py

```diff
@@ -1415,6 +1415,24 @@ def pics_guard(self, pics_condition: bool):
             self.mark_current_step_skipped()
         return pics_condition
 
+    async def attributes_guard(self, attribute_condition: bool):
+        """Similar to pics_guard above, except checks a condition and if False marks the test step as skipped and
+           returns False using attributes against attributes_list, otherwise returns True.
+           For example can be used to check if a test step should be run:
+
+              self.step("1")
+              if self.attribute_guard(condition1_needs_to_be_true_to_execute):
+                  # do the test for step 1
+
+              self.step("2")
+              if self.attribute_guard(condition2_needs_to_be_false_to_skip_step):
+                  # skip step 2 if condition not met
+           """
+        attr_condition = await asyncio.wait_for(should_run_test_on_endpoint(self, has_attribute(attribute_condition)), timeout=60)
```

Comment:
The should_run_test_on_endpoint function is going to do a wildcard read on every call.

Given that we're moving now towards doing a lot of checks based on the device conformance, I think it would make sense to move that wildcard read up to the setup_class function and have it stored as self.wildcard and use that. Then you can just call has_attribute directly here (and swap over the should_run... to use the stored wildcard)

Reply:
Actually, we can remove some duplication with the basic composition tests if you do the read then save that as self.endpoints / self.endpoints_tlv then swap over from using wildcard.

Reply:
Hi Cecille,
I have updated the above method for the attributes_guard function as discussed during our call this afternoon, thank you